### DISCUSSION TITLE
Can we integrate this SX1262 radio code in MCCI LMIC?

### DISCUSSION THREAD

**Discussion by cyberman54** [ngraziano/LMICPP-Arduino#20] [Ideas]
Would it be possible to make a PR to add this code for supporting SX1262 to MCCI LMIC lib?

**Comment by ngraziano** [comment]
To add SX1262 to MCCI need some works (I think more than 30h of work)  and lately I lack motivation for development and I only have one board with SX1262 and it is with an arduino pro mini (a little small for MCCI version).

I made a lot of modification to lmic before adding sx1262 radio and I use C++ class instead of C code so create a PR is not easy.

The step I see to add SX1262 in MCCI code is:
- complete documentation ([doc\RadioDriver.md](https://github.com/mcci-catena/arduino-lmic/blob/master/doc/RadioDriver.md)) with the calls to `radio_init` and `radio_irq_handler_v2` and add some `#ifndef` for this functions
- find a way to split the check of the pins "dio" not the same way for different radio (sx1262 have a busy pin that must be check before sending command but only need one dio pin (i am not sure how to make it clean)
- then create an new driver (by looking at my code https://github.com/ngraziano/LMICPP-Arduino/blob/master/src/lmic/radio_sx1262.cpp  and https://github.com/LacunaSpace/basicmac/blob/master/lmic/radio-sx126x.c which is more C style)
